lib/cocoapods/native_target_extension.rb

```diff
@@ -0,0 +1,51 @@
+module Pod
+  class Project
+    # Adds a dependency on the given metadata cache.
+    #
+    # @param  [AbstractTarget] target
+    #         The parent target used to add a cached dependency.
+    #
+    # @param  [MetadataCache] metadata
+    #         The metadata holding all the required metadata to construct a target as a dependency.
+    #
+    # @return [void]
+    #
+    def add_cached_dependency(target, metadata)
+      return unless dependency_for_cached_target(target, metadata)
+      container_proxy = target.project.new(Xcodeproj::Project::PBXContainerItemProxy)
+
+      subproject_reference = target.project.reference_for_path(metadata.container_project_path)
+      raise ArgumentError, "add_dependency received target (#{target}) that belongs to a project that is not this project (#{self}) and is not a subproject of this project" unless subproject_reference
+      container_proxy.container_portal = subproject_reference.uuid
+
+      container_proxy.proxy_type = Constants::PROXY_TYPES[:native_target]
+      container_proxy.remote_global_id_string = metadata.native_target_uuid
+      container_proxy.remote_info = metadata.target_label
+
+      dependency = target.project.new(Xcodeproj::Project::PBXTargetDependency)
+      dependency.name = metadata.target_label
+      dependency.target_proxy = container_proxy
+
+      target.dependencies << dependency
+    end
+
+    # Checks whether this target has a dependency on the given target.
```

Comment:
nit: this documentation does not match the method return. I had thought it was a bool to begin with and I almost asked to append a `?` to the method name

Reply:
It should be a bool. Adding a `?` to name

Reply:
Cool, update the doc to specify `[Boolean]` in return type